privateSBOMExchange/src/petra/lib/models/__init__.py

```diff
@@ -1,13 +1,304 @@
+        self.policy=None
+    def accept(self, visitor):
+        return visitor.visit_field_node(self)
+
+
+class ComplexNode(Node):
+    """Represents a package node that can have multiple children of type FieldNode.
+
+    This node represents a dependency that does not have an available SBOM tree in the database.
+
+    Attributes
+    ----------
+    field_name : str
+        "PackageName".
+    field_value :str
+        The package name.
+    encrypted_and_hashed_data : bytes or None
+        The encrypted and hashed representation of the package data, initially set to None.
+    children : List[FieldNode]
+        A list of child FieldNode instances.
+    """
+    def __init__(self, type_name:str,type_value:str, children:List[FieldNode]):
+        """Initialize a PackageNode.
+
+        Parameters
+        ----------
+        package_name : str
+            The name of the package.
+        children : List[FieldNode]
+            A list of FieldNode instances representing the fields of the package.
+        """
+        self.metadata_type_name:str= type_name  # Store data as string
+        self.metadata_type_value:str=type_value
+        self.encrypted_data=None #store encrypted data when visited by Encrypt visitor
+        self.hash=None #store hashed data when visited by Merkle visitor
+        self.children = children
+
+    def accept(self, visitor):
+        return visitor.visit_complex_node(self)
+
+
+class SbomNode(Node):
+    """Represents the root node of a Software Bill of Materials (SBOM) tree.
+
+    This node can have three types of children, all of which are derived from the Node class:
+    1. **FieldNode**: Represents a field in an SBOM document.
+    2. **ComplexNode**: Represents a complex node that has multiple fields.
+    3. **SbomNode**: Represents a package dependency that does have an available SBOM tree in the database.
+
+    Attributes
+    ----------
+    SBOM_name : str
+        The name or identifier of the SBOM.
+    encrypted_data : bytes or None
+        The encrypted verison of the data, initially set to None.
+    hash : bytes or None
+        The hashed version of the data, initially set to None.
+    children : List[Node]
+        A list of child nodes, which can be FieldNode, PackageNode, or other SbomNode instances.
+    """
+    def __init__(self,name:str,purl, children:List[Node]):
+        """Initialize an SbomNode.
+
+        Parameters
+        ----------
+        name : str
+            The name of the SBOM.
+        children : List[Node]
+            A list of child nodes that can be FieldNode, PackageNode, or other SbomNode instances.
+        """
+        self.SBOM_name=name
+        self.encrypted_data=None #store encrypted data when visited by Encrypt visitor
+        self.hash=None #store hashed data when visited by Merkle visitor
+        self.children = children  # Hold the children directly
+        self.signiture=None
```

Comment:
typo: this should be `signature`.

Reply:
Can we push a fix to this typo pls?

Reply:
@marcelamelara @SantiagoTorres  I just pushed the edits